src/main/java/com/cloudbees/jenkins/support/impl/ItemsContent.java

```diff
@@ -0,0 +1,202 @@
+                    Map<String,ItemsContent.Stats> containerStats = new HashMap<String,ItemsContent.Stats>();
+
+                    long startTime = System.currentTimeMillis();
+                    for (Item i : jenkins.getAllItems()) {
+                        String key = i.getClass().getName();
+                        Integer cnt = containerCounts.get(key);
+                        containerCounts.put(key, cnt == null ? 1 : cnt + 1);
+                        if (i instanceof Job) {
+                            Job<?,?> j = (Job) i;
+                            int builds = 0;
+                            Iterator buildsIterator = j.getBuilds().iterator();
+                            builds = Iterators.size(buildsIterator);
+                            jobTotal.add(builds);
+                            ItemsContent.Stats s = jobStats.get(key);
+                            if (s == null) {
+                                jobStats.put(key, s = new ItemsContent.Stats());
+                            }
+                            s.add(builds);
+                        }
+                        if (i instanceof ItemGroup) {
+                            ItemsContent.Stats s = containerStats.get(key);
+                            if (s == null) {
+                                containerStats.put(key, s = new ItemsContent.Stats());
+                            }
+                            s.add(((ItemGroup) i).getItems().size());
+                        }
+                    }
+                    long endTime = System.currentTimeMillis();
+                    LOGGER.log(Level.FINE, "Time to compute all the build is {0}", endTime - startTime);
+                    out.println("Item statistics");
+                    out.println("===============");
+                    out.println();
+                    for (Map.Entry<String,Integer> entry : containerCounts.entrySet()) {
+                        String key = entry.getKey();
+                        out.println("  * `" + key + "`");
+                        out.println("    - Number of items: " + entry.getValue());
+                        ItemsContent.Stats s = jobStats.get(key);
+                        if (s != null) {
+                            out.println("    - Number of builds per job: " + s);
+                        }
+                        s = containerStats.get(key);
+                        if (s != null) {
+                            out.println("    - Number of items per container: " + s);
+                        }
+                    }
+                    out.println();
+                    out.println("Total job statistics");
+                    out.println("======================");
+                    out.println();
+                    out.println("  * Number of jobs: " + jobTotal.n());
+                    out.println("  * Number of builds per job: " + jobTotal);
+                }
+            });
+        }
+    }
+
+    private static class Stats {
```

Comment:
I'm not a big fan of duplicating this class. Do you think it makes sense to move it to an external class to be then used anywhere necessary?

Reply:
Yes, I will submit the changes

Reply:
Sorry, the class is not used in `AboutJenkins` anymore, so I just removed it from there. No need to create a new class only for this.